test/compliment/sources/t_class_members.clj

```diff
@@ -207,3 +207,57 @@
 
   (fact "static class members have docs"
     (src/static-member-doc "Integer/parseInt" (-ns)) => (checker string?)))
+
+(def java-version (-> (System/getProperty "java.version")
+                      (str/split #"\.")
+                      first
+                      read-string))
+
+(deftest literals-inference-test
+  (testing "Vector literals"
+    (let [candidates-count (count (src/members-candidates "." (-ns) (ctx/cache-context
+                                                                     "(__prefix__ [])")))
+          {:keys [major minor]} *clojure-version*
+          expected (if (and (= 1 major)
+                            (< minor 12))
+                     18
+                     19)]
```

Comment:
Sorry, you are overdoing it. Your changes to Compliment don't depend on the exact public methods of PersistentVector or other classes, they transcend the exact versions of Clojure and Java. Just make a minimal test that proves that context performs the filtering, and that's it. There are plenty of tests like that already, take any as an example.

Reply:
Done, I believe!